pkg/markers/register.go

```diff
@@ -17,13 +20,20 @@ type definitionWithHelp struct {
 }
 
 type Registry struct {
-	mRegistry *markers.Registry
+	mRegistry               *markers.Registry
+	ignoredKubeMarkersRegex *regexp.Regexp
 }
 
-func NewRegistry() (*Registry, error) {
+func NewRegistry(ignoredKubeMarkers []string) (*Registry, error) {
+	var ignoredKubeMarkersRegexp *regexp.Regexp
+	if len(ignoredKubeMarkers) > 0 {
+		toIgnore := strings.Join(ignoredKubeMarkers, "|")
+		ignoredKubeMarkersRegexp = regexp.MustCompile(fmt.Sprintf("(?:%s)", toIgnore))
+	}
 	mReg := &markers.Registry{}
 	r := &Registry{
-		mRegistry: mReg,
+		mRegistry:               mReg,
+		ignoredKubeMarkersRegex: ignoredKubeMarkersRegexp,
```

Comment:
I'd prefer we not mess with this registry and instead localize the hack to ignore markers within openapiGenerator.go when we parse the markers.

I don't like the idea of parsing markers that should be ignored but ignoring them during processing. Instead we should just not consider that specific marker when we compute the validation rules.

Reply:
I agree, modified the PR to use the approach we discussed offline, which is to prevent the marker from being registered if it is ignored.